src/exodus/sigmadb.cpp

```diff
@@ -0,0 +1,485 @@
+    int nBlock)
+{
+    // Logic:
+    // Get next group id and index for new pubkey by get last group id and amount of coin in group
+    // If the count is equal to limit then move to new group
+    // Record mint by key `0<prob_id><denom><group_id><idx>` with value `<GroupElement><int>`
+    // Record the key `0<prob_id><denom><group_id><idx>` as value of `1<sequence>`
+    // Record Last group ID
+    // Record Mint count for group
+
+    auto lastGroup = GetLastGroupID(propertyID, denomination);
+    auto mints = GetMintCount(propertyID, denomination, lastGroup);
+    uint32_t nextIDx = mints;
+
+    if (mints >= MaxCoinsPerGroup) {
+        lastGroup++;
+        nextIDx = 0;
+    }
+
+    auto rawMintKey = CreateMintKey(propertyID, denomination, lastGroup, nextIDx);
+    leveldb::Slice mintKey = GetSlice(rawMintKey);
+
+    auto commitment = pubKey.GetCommitment();
+
+    std::vector<uint8_t> buffer(commitment.memoryRequired() + sizeof(int));
+    auto ptr = buffer.data();
+    ptr = commitment.serialize(ptr);
+    std::memcpy(ptr, &nBlock, sizeof(nBlock));
+
+    pdb->Put(writeoptions, mintKey, GetSlice(buffer));
+
+    // Store key
+    RecordMintKeyIndex(mintKey);
+
+    // Store last group id
+    RecordLastGroupID(propertyID, denomination, lastGroup);
+
+    // Store mint count
+    RecordMintCount(propertyID, denomination, lastGroup, nextIDx + 1);
+}
+
+std::vector<exodus::SigmaPublicKey> CMPMintList::GetAnonimityGroup(
+    uint32_t propertyID, uint32_t denomination, uint32_t groupID, uint32_t idx)
+{
+    std::vector<exodus::SigmaPublicKey> pubKeys;
+
+    auto firstKey = CreateMintKey(propertyID, denomination, groupID, 0);
+
+    auto it = NewIterator();
+    it->Seek(GetSlice(firstKey));
+
+    uint32_t mintPropID, mintDenom, mintGroupID, mintIDx;
+
+    for (uint32_t i = 0; i < idx && it->Valid(); i++, it->Next()) {
+        if (!ParseMintKey(it->key(), mintPropID, mintDenom, mintGroupID, mintIDx) ||
+            mintPropID != propertyID ||
+            mintDenom != denomination ||
+            mintGroupID != groupID) {
+            LogPrintf("%s() : read invalid key", __func__);
+            break;
+        }
+
+        if (mintIDx > idx) {
+            LogPrintf("%s() : read exceed query index", __func__);
+            break;
+        }
+
+        exodus::SigmaPublicKey pub;
+        std::tie(pub, std::ignore) = ParseMint(it->value().ToString());
+
+        if (pub.GetCommitment().isMember()) {
+            pubKeys.push_back(pub);
+        }
+    }
+
+    return pubKeys;
+}
+
+void CMPMintList::DeleteAll(int nBlock)
+{
+    auto nextSequence = GetNextSequence();
+    if (nextSequence <= 0) {
+        // No mint to delete
+        return;
+    }
+
+    auto lastSequence = nextSequence - 1;
+
+    auto rawLastSequenceKey = CreateIndirectMintKey(lastSequence);
+
+    auto it = NewIterator();
+    it->Seek(GetSlice(rawLastSequenceKey));
+
+    std::vector<std::string> keyToDeletes;
+    uint64_t lastDeleted = nextSequence;
```

Comment:
I think rename `lastDeleted` to `remaining` will be more easy to understand.

Reply:
removed